learning_journal/tests.py

```diff
@@ -1,65 +1,249 @@
+     from the ENTRIES list.
+    """
+    for each in TEST_ENTRIES:
+        model = Entry(title=each["title"], creation_date=each["creation_date"], body=each["body"])
+        dummy_request.dbsession.add(model)
+
+
+def test_database_empty_but_exists(db_session):
+    """Test that app connects to database, which is empty here."""
+    assert len(db_session.query(Entry).all()) == 0
+
+
+def test_adding_model(db_session):
+    """Test the model gets added to the database."""
+    assert len(db_session.query(Entry).all()) == 0
+    model = Entry(title="new title for new test", body="new body for new test",
+        creation_date=datetime.datetime(2017, 1, 1, 0, 0), id="8675309")
+    db_session.add(model)
+    assert len(db_session.query(Entry).all()) == 1
+
+
+def test_my_view(dummy_request):
+    """Test the homepage view returns data from database."""
+    from .views.default import my_view
+    dummy_request.dbsession.add(Entry(title="one", id='1'))
+    result = my_view(dummy_request) # views commit changes to the DB
+    assert result["entries"][0].title == "one"
+
+
+def test_homepage_returns_empty_when_empty(dummy_request):
+    """Test that the list view returns no objects in the expenses iterable."""
+    from learning_journal.views.default import my_view
+    result = my_view(dummy_request)
+    assert len(result["entries"]) == 0
+
+def test_homepage_returns_existing_entries(dummy_request, add_models):
+    """Test that the list view serves up journal entries."""
+    from learning_journal.views.default import my_view
+    result = my_view(dummy_request)
+    assert len(result["entries"]) == len(TEST_ENTRIES)
+
+
+def test_detail_view(dummy_request, db_session):
+    """Test the detail view."""
+    from .views.default import detail
+    dummy_request.method = "POST"
+    dummy_request.POST["title"] = "This is a Title"
+    dummy_request.POST["body"] = "And this is the body. TEST!"
+    dummy_request.matchdict['id'] = '1'
+    result = detail(dummy_request)
+    entry = dummy_request.dbsession.query(Entry).get(1)
+    assert result['entry'] == entry
+
+
+def test_detail_page_returns_empty_when_empty(dummy_request):
+    """Test that the detail view returns no objects when database empty."""
+    from learning_journal.views.default import detail
+    req = dummy_request
+    req.matchdict = {"id": "1"}
+    result = detail(req)
+    assert result["entry"] is None
+
+
+def test_edit_page_updates_db(dummy_request, add_models):
+    """Test that the edit view edits entries in the database."""
+    from learning_journal.views.default import edit
+    req = dummy_request
+    req.matchdict = {"id": "3"}
+    req.method = "POST"
+    req.POST["title"] = "new thing"
+    req.POST["creation_date"] = datetime.datetime(2017, 2, 24, 0, 0)
+    req.POST["body"] = "and more new things"
+    try:
+        edit(req)
+    except:
+        new_title = dummy_request.dbsession.query(Entry).get(3).title
+        assert new_title == "new thing"
```

Comment:
I'm not sure what I was going for there; got rid of the try-except, dedented 175-176, and happy to see it working